x/snapshot/exported/types.go

```diff
@@ -160,9 +160,22 @@ func (v ValidatorIllegibility) FilterIllegibilityForNewKey() ValidatorIllegibili
 	return v & mask
 }
 
-// FilterIllegibilityForSigning filters the illegibility to only leave those ones related to handling of signing
-func (v ValidatorIllegibility) FilterIllegibilityForSigning() ValidatorIllegibility {
-	return v & ^NoProxyRegistered
+// FilterIllegibilityForTssSigning filters the illegibility to only leave those ones related to handling of signing
+func (v ValidatorIllegibility) FilterIllegibilityForTssSigning() ValidatorIllegibility {
+	mask := None
+
+	for _, illegibility := range GetValidatorIllegibilities() {
+		mask |= illegibility
+	}
+
+	return v & mask
+}
```

Comment:
can't we just return `v` here?

Reply:
It's more of a sanity thing, just to make sure v doesn't have any bit set to 1 beyond the defined range.